### ISSUE TITLE
[Bug] - Deadlock found when trying to get lock on DataObject import 

### ISSUE THREAD

**Issue by SamyMP** [pimcore/pimcore#15150]
Hi, 

For your information, we are facing the exact same problem with the `object_meatada` table on one of our object. 

What happens is, we are trying to populate a fresh install of pimcore with a little bit more than 100 000 DataObjects through the DataImporter Bundle. These DataObjects have multiple relation fields. Some of these fields are filled through the DataImporter, but others are filled using an EventListener (using the `PreSaveEvent` of the DataImporter) because what we are trying to achieve is not possible natively with the DataImporter.

When we execute the import in parallel with multiple process (ie. 10), a lot of objects couldn't be created because of this error :
```
Next Doctrine\DBAL\Exception\DeadlockException: An exception occurred while executing 'INSERT INTO object_metadata_OBJECT (o_id, dest_id, fieldname, ownertype, ownername, index, position, type, column, data) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE o_id = ?, dest_id = ?, fieldname = ?, ownertype = ?, ownername = ?, index = ?, position = ?, type = ?, column = ?, data = ?;' with params [my params]:

SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction in /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:51
```

It failed even though it retried, according to the message.

The params (my enveloppes) are good since, to try to debug this, we imported all of our objects by executing the parallel process with 1 process (took 6 hours, but no errors). It is clearly linked to the parallelization of the import.

I also have to mention that we were able to import everything by commenting our event in the DataImporter, with multiple processes (again, 10). So the events in the DataImporter seems to be a part of the problem as well in our case. 

Lastly, the problem doesn't seem to appear when updating already created objects. It only affects the creation.

Should I create a new issue with this problem, or is it also part of this problem ?

_Originally posted by @SamyMP in https://github.com/pimcore/pimcore/issues/15064#issuecomment-1539650108_

**Comment by dvesh3** [comment]
Fixed by #15391

**Comment by dvesh3** [comment]
Please see https://github.com/pimcore/pimcore/pull/15591

**Comment by dvesh3** [comment]
Fixed by #15746

**Comment by SamyMP** [comment]
Thank you ! 🙂
